crates/storage/src/connection/transaction.rs

```diff
@@ -225,6 +225,35 @@ pub(super) fn transaction_data_for_block(
     Ok(Some(data))
 }
 
+pub(super) fn transaction_hashes_for_block(
+    tx: &Transaction<'_>,
+    block: BlockId,
+) -> anyhow::Result<Option<Vec<TransactionHash>>> {
+    let Some((_, block_hash)) = tx.block_id(block)? else {
+        return Ok(None);
+    };
+
+    let mut stmt = tx
+        .inner()
+        .prepare("SELECT hash FROM starknet_transactions WHERE block_hash = ? ORDER BY idx ASC")
+        .context("Preparing statement")?;
+
+    let mut rows = stmt
+        .query(params![&block_hash])
+        .context("Executing query")?;
+
+    let mut data = Vec::new();
+    while let Some(row) = rows.next()? {
+        let hash = row
+            .get_transaction_hash("hash")
+            .context("Fetching transaction hash")?;
+
+        data.push(hash);
+    }
```

Comment:
You can also use the `query_map` instead; I believe you can then do some variation of
```suggestion
        .query_map(
            params![&block_hash], 
            |row| row.get_transaction_hash("hash")
        )
        .context("Executing query")?
        .collect()
```